### PR TITLE
contracts-stylus: darkpool: add `Ownable` & `Pausable` controls

### PR DESCRIPTION
**Pull request by akirillo**
This PR makes the `DarkpoolContract` `Ownable` and `Pausable`. Now, all setter methods can be "paused" and "unpaused" by the contract owner, a role which can be transferred. Notably, we do not include events for ownership transfer or pausing, nor do we include an explicit `renounce_ownership` method. This is because these features take up binary size that is best left for other potential controls, and can be brought in via an upgrade once size constraints slacken.

**Testing:**
Tests have been added asserting proper `Ownable` and `Pausable` functionality, which pass (with the exception of asserting that `process_match_settle` works as intended when paused / unpaused, because completing the e2e tests for that is a separate workstream).

### CODE REVIEW

File: contracts-utils/src/conversion.rs [hunk 1]
```diff
@@ -43,3 +48,27 @@ pub fn to_contract_vkey(
...
+            .y
+            .into_iter()
+            .map(Scalar::new)
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap(),
+    };
+
+    CircuitPublicSigningKey { x, y }
+}
```
Comment (akirillo) [inline contracts-utils/src/conversion.rs:74]: this was moved, no need for review

File: contracts-utils/src/proof_system/test_data.rs [hunk 2]
```diff
@@ -45,6 +53,58 @@ pub fn dummy_statement<R: RngCore + CryptoRng, S: CircuitBaseType>(rng: &mut R)
...
+        ]
+        .concat(),
+    );
+
+    let public_inputs_signature = Bytes::from(
+        hash_and_sign_message(&signing_key, &shares_commitment.serialize_to_bytes()).to_vec(),
+    );
+
+    Ok((proof, contract_statement, public_inputs_signature))
+}
```
Comment (akirillo) [inline contracts-utils/src/proof_system/test_data.rs:106]: this was moved, no need for review

File: contracts-stylus/src/contracts/darkpool.rs [hunk 3]
```diff
@@ -275,6 +335,32 @@ impl DarkpoolContract {
...
+    pub fn _check_owner<S: TopLevelStorage + Borrow<Self>>(storage: &S) {
+        assert_eq!(storage.borrow().owner.get(), msg::sender())
+    }
+
+    // ------------
+    // | PAUSABLE |
+    // ------------
+
+    /// Checks whether the darkpool's paused status matches the given `paused` value
+    pub fn _check_paused_status<S: TopLevelStorage + Borrow<Self>>(storage: &S, paused: bool) {
```
Comment (joeykraut) [inline contracts-stylus/src/contracts/darkpool.rs]: "check" here seems to imply that we'd return a bool, can we break this out into `_assert_paused` and `_assert_unpaused`? I assume those are reasonably small code size diffs, and that would make the actual enforced semantics much clearer at the callsite. Esp for something as critical as this.

File: integration/src/tests.rs [hunk 4]
```diff
@@ -378,6 +379,165 @@ pub(crate) async fn test_initializable(
...
+        &contract_with_dummy_owner,
+        UNPAUSE_METHOD_NAME,
+        (),
+    )
+    .await?;
+
+    Ok(())
+}
+
+pub(crate) async fn test_pausable(
```
Comment (joeykraut) [inline integration/src/tests.rs:453]: can we test somewhere in here that pausing is only allowed by the owner? might already be implicitly tested but if not that'd be a good base to cover

**Review by joeykraut — APPROVED** [review]
nicely done

File: integration/src/tests.rs (hunk 4 above)
Comment (akirillo) [reply integration/src/tests.rs:453]: yup, that's tested on lines 440 / 442

**Pull request merged**